### DISCUSSION TITLE
About YouTube playback errors - YouTube changed some things to get working streams

### DISCUSSION THREAD

**Discussion by wb9688** [TeamNewPipe/NewPipe#11260] [Announcements]
# TLDR: We will work on a fix which should come in the next days.

## What is happening?

For the past weeks/months, YouTube has started to A/B test the requirement of an anti-bot check on streaming URLs from its HTML5 clients (desktop, mobile and TV websites), a `pot` URL query parameter, which leads to invalid responses (HTTP response code 403) after some time, like on the Android app if you do not pass integrity checks (this happened a few times to NewPipe and reVanced which works around the issue). This anti-bot check is hard to implement and requires a full browser environment. See #11191.

For the past weeks or months too, YouTube started to require the parameter which was before responsible of throttling streaming URLs (`n` query parameter of streaming URLs) if you do no pass the correct value, computed by the YouTube's JavaScript player. This requirement leads so now to 403 HTTP responses too.

Since 2nd week of July 2024, YouTube has started to roll out a new JavaScript player which changes how this function can be extracted. The function to return the correct value cannot be extracted currently by the extractor, leading to instant 403 HTTP responses. See #11255. This has been fixed in [the 0.27.1 release](https://github.com/TeamNewPipe/NewPipe/releases/tag/v0.27.1).

## What should I do know?

Wait and do not open duplicate issues or discussions on GitHub about this issue. We will work on a fix which will work around the issue. Livestreams should be still playable, unless you're affected by the `Sign to confirm that you're not a bot` error (see above).

## What are the limitations of the workaround?

The workaround will not apply on age-restricted videos, which may become unplayable until we know how to implement the anti-bot check. Clients of NewPipe Extractor will be not able to play 8K streams (NewPipe doesn't support this resolution yet).

### I am affected by the Sign to confirm that you're not a bot error. Is there something I can do?

Unless you are able to change your IP address (by using a VPN, a proxy, or being able to change your router's one), nothing in NewPipe for now. See #11139.

You can also try to play videos with a Google account outside of NewPipe or watch videos with embeds if the creator(s)/right(s) owner(s) allows to do so.

Fixing this issue without logging in is related to the anti-bot check described above.

### Before the HTTP 403 issue which started yesterday, I wasn't able to get a YouTube 720 MPEG-4 video stream with audio for external players for a few weeks. Is it normal?

Yes, YouTube removed this stream (stream number "itag" 22). See #11214 and yt-dlp/yt-dlp#10206.

### You were spoofing the Android app to get streams, why it doesn't work now?

Like written in [the changelog of 0.27.0 on GitHub](https://github.com/TeamNewPipe/NewPipe/releases/tag/v0.27.0), it doesn't work as YouTube patched all known workarounds to get streams from this client. NewPipe Extractor so falls back to streams from the desktop website, affected by the changes described above.

**Comment by KENCHASEN** [comment]
My Newpipe app has problems downloading please help 🔥🚨🚨🚨🚨

**Comment by Niterux** [comment]
Will this workaround reduce my performance? My phone can't really handle web browsers 😦

**Comment by SGR21gtb** [comment]
Uhh now we need a fix for sign in to confirm you're not a bot. Thanks, MePipe!
